src/context/ProfileContext.jsx

```diff
@@ -0,0 +1,51 @@
+import { createContext, useContext, useEffect, useState } from 'react';
+import { getProfile } from '../services/profiles';
+
+const ProfileContext = createContext();
+
+function ProfileProvider({ children }) {
+  const [loading, setLoading] = useState(false);
+  const [profile, setProfile] = useState({
+    name: '',
+    email: '',
+    bio: '',
+    birthday: '',
+  });
+
+  useEffect(() => {
+    const fetchProfile = async () => {
+      setLoading(true);
+      try {
+        const resp = await getProfile();
+
+        if (resp.length > 0) {
+          setProfile(resp[0]);
+        }
+      } catch (error) {
```

Comment:
This `catch` probably isn't necessary as `getProfile` does not return and error if no profile is found without the `.single()`